paddle/testing/paddle_gtest_main.cc

```diff
@@ -61,29 +61,30 @@ int main(int argc, char** argv) {
   undefok.push_back("initial_cpu_memory_in_mb");
 #endif
 
+  std::string env_string = "--tryfromenv=";
   if (envs.size() > 0) {
-    std::string env_string = "--tryfromenv=";
     for (auto t : envs) {
       env_string += t + ",";
     }
     env_string = env_string.substr(0, env_string.length() - 1);
-    new_argv.push_back(strdup(env_string.c_str()));
+    new_argv.push_back(const_cast<char*>(env_string.c_str()));
```

Comment:
It is not recommended to use `const_cast`, could you use another way?

Reply:
I prepared other solution (without const_cast)